### ISSUE TITLE
Add learnable image resizing

### ISSUE THREAD

**Issue by cisprague** [google/flax#5000]
With images, it's often useful to have a layer that maps you from one spatial shape to another (i.e., upsampling or downsampling), e.g. as a building block in U-Net. This can be achieved through [interpolation](https://github.com/jax-ml/jax/blob/01f9d46aef29ac3dfa479709aaab64479731af85/jax/_src/image/scale.py#L302) or convolutional layers ([Conv](https://github.com/google/flax/blob/771eadb1986e59a5a2dd82a14188fd1cd7b3849f/flax/nnx/nn/linear.py#L600) or [ConvTranspose](https://github.com/google/flax/blob/771eadb1986e59a5a2dd82a14188fd1cd7b3849f/flax/nnx/nn/linear.py#L873)) – where, with a given kernel and a dilation of 1, stride and padding can be used to achieve the mapping (while minimizing padding).

Would it be useful to have an `nnx.Module` like the following?

```python
class ConvScale(nnx.Module):

    def __init__(
        self, *,
        in_shape: Sequence[int], 
        out_shape: Sequence[int],
        kernel: Sequence[int] | str,
        rngs: nnx.Rngs
    ):
        '''
        Args:
            in_shape (Tuple[int]): input shape with last dimension 
                as features, and rest as spatial dimensions.
            out_shape (Tuple[int]): out shape with last dimension 
                as features, and rest as spatial dimensions.
            kernel (Tuple[int]): kernel size per spatial dimension.
                If str, we use standard interpolation with chosen str method.
        '''
```

Imagine you have a sequence of `T` images (e.g., from a video):

```python
assert x.shape == (T, H, W, C)
module = ConvScale(
    in_shape=(T, H, W, C),
    out_shape=(T_new, H_new, W_new, C_new),
    kernel=(k, k, k),
    # kernel='linear',
    rngs=nnx.Rngs(0)
)
assert module(x).shape == (T_new, H_new, W_new, C_new)
```

I wonder if it would be useful for a PR?

**Comment by cgarciae** [comment]
Hey @cisprague! We try to maintain a very small set of layers--ones that are considered canonical--and users can maintain more specialized ones on their projects. ConvScale seems to be a combination of `jax.image.resize` + `Conv` so users can just code this on their end. However, maybe you can share you implementation on the Discussions page, that would be an equally appreciated contribution for the community.

**Comment by cisprague** [comment]
Sounds good. It was just that the formula you need to use to do `(A, A, C) -> (B, B, C)` with a given kernel size in a `Conv` layer was not obvious for me to find on the internet. A utility for this in JAX would've been useful to me at the time.

**Comment by vfdev-5** [comment]
I think we can close this issue as resolved. @cisprague feel free to reopen if we need to discuss more this topic. Thanks!
